### PR TITLE
cleanup(GCS+gRPC): remove hybrid client

### PR DESCRIPTION
**Pull request by coryan**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13820)

### CODE REVIEW

File: google/cloud/storage/grpc_plugin.cc [hunk 1]
```diff
@@ -32,13 +31,8 @@ google::cloud::storage::Client DefaultGrpcClient(Options opts) {
...
     return google::cloud::storage::Client(std::move(opts));
   }
-  auto stub = [&]() -> std::unique_ptr<storage_internal::GenericStub> {
-    if (config == "metadata" || config.empty()) {
-      opts = google::cloud::storage_internal::DefaultOptionsGrpc(opts);
-      return std::make_unique<storage_internal::GrpcStub>(opts);
-    }
-    return std::make_unique<storage_internal::HybridStub>(opts);
-  }();
+  opts = google::cloud::storage_internal::DefaultOptionsGrpc(opts);
```
Comment (devbww) [inline google/cloud/storage/grpc_plugin.cc:34]: `std::move(opts)`?

File: google/cloud/storage/grpc_plugin_test.cc [hunk 2]
```diff
@@ -91,18 +91,19 @@ TEST(GrpcPluginTest, NoneConfigCreatesCurl) {
               ElementsAre("RestStub", "StorageConnectionImpl"));
 }
 
-TEST(GrpcPluginTest, MediaConfigCreatesHybrid) {
+TEST(GrpcPluginTest, MediaConfigCreatesMetadata) {
   // Explicitly disable logging, which may be enabled by our CI builds.
   auto logging =
       ScopedEnvironment("CLOUD_STORAGE_ENABLE_TRACING", absl::nullopt);
+  // We used to support a "media" configuration where only media operations
```
Comment (devbww) [inline google/cloud/storage/grpc_plugin_test.cc:98]: Nit: Do we have to say what we used to do?

**Review by devbww — APPROVED** [review]

File: google/cloud/storage/grpc_plugin.cc (hunk 1 above)
Comment (coryan) [reply google/cloud/storage/grpc_plugin.cc]: Done.

File: google/cloud/storage/grpc_plugin_test.cc (hunk 2 above)
Comment (coryan) [reply google/cloud/storage/grpc_plugin_test.cc]: Removed, and refactored the test some more.

**Pull request merged**